### ISSUE TITLE
Detect incorrectly positioned return schema

### ISSUE THREAD

**Issue by frenchy64** [clj-kondo/clj-kondo#2374]
- [X] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**Is your feature request related to a problem? Please describe.**

It's easy to forget where the return schema goes in `schema.core` macros. The incorrect position silently fails to be checked.

```clojure
;; good
(s/defn f :- s/Str [])
;; bad
(s/defn f [] :- s/Str)
```

**Describe the solution you'd like**

A warning if `:-` is detected after the params vector in any schema macro.

**Describe alternatives you've considered**

Schema itself could warn during expansion.

**Additional context**

We should also consider other forms when testing this linter.

```clojure
(s/defn f (:- s/Str []))
(s/defn f :- s/Str ([]))
(s/defn f ([] :- s/Str))
```

There is some overlap here with an "unused statements" linter, further investigation might be needed if there is one already or if it could be improved more generally.

I can PR this if interested.

**Comment by borkdude** [comment]
I'm not using schema a lot but if you find this useful and want to provide a PR, sure!

**Comment by frenchy64** [comment]
@borkdude ready for a review https://github.com/clj-kondo/clj-kondo/pull/2375

**Comment by borkdude** [comment]
I'll have a look soon-ish!

**Comment by frenchy64** [comment]
Commit: https://github.com/clj-kondo/clj-kondo/commit/30ab602bbfb0381b164c73f67aaef26727a5bb6d
